src/Sylius/Behat/Context/Ui/ProductContext.php

```diff
@@ -52,4 +82,26 @@ public function iShouldNotBeAbleToAccessProduct(ProductInterface $product)
 
         expect($this->productShowPage->isOpen(['product' => $product]))->toBe(false);
     }
+
+    /**
+     * @When I delete the :product product
+     */
+    public function iDeleteProduct(ProductInterface $product)
+    {
+        $this->sharedStorage->set('product', $product);
```

Comment:
I've just checked. It does not dissapear, so it can be set at the end of this function just like @Zales0123 suggested. :)